core/sql/optimizer/RelCache.cpp

```diff
@@ -418,6 +418,10 @@ void GroupByAgg::generateCacheKey(CacheWA &cwa) const
     groupExpr_.rebuildExprTree(ITM_ITEM_LIST);
   if (grpExpr) { 
     cwa += " gBy:"; 
+
+    if (isRollup())
+      cwa += " roll:";
+
```

Comment:
I think we also need to add the rollupGroupExprList() to the cache key. If we rebuild the list above from a ValueIdSet on line 418 above, it is probably going to be in the same order, regardless whether it was ROLLUP(a,b) or ROLLUP(b,a).